apicast/src/oauth/keycloak.lua

```diff
@@ -248,7 +262,8 @@ function _M:get_token(service, client)
   -- call Keycloak authorize
   local url = self.config.token_url
 
-  local res = http_client.post(url, req_body)
+  local headers = ngx.req.get_headers()
+  local res = http_client.post(url, ngx.req.get_post_args(), { headers = headers })
```

Comment:
Re: testing, yes, there is a test that checks the authorization header is passed ([link](https://github.com/3scale/apicast/pull/336/files/8cd98a1f55125a836cd94603cca2223460d0d716#diff-21684f4035d756447c84d4b9d2a72d6fR195), the previous one was not correct), but I don't know how to check that other headers were not passed. Could be an integration test, but maybe not so necessary?